consumption.tex

```diff
@@ -0,0 +1,89 @@
+
+In the interest of keeping this implementation generic we assume that the user receives a token in return for his burn. However, instead of minting a token, the target cryptocurrency could allow the burn verifier contract to mint native cryptocurrency for any user who successfully claims an event. This would allow the target cryptocurrency to be bootstrapped entirely though burning as desired.
+
+%The workflow is as follows. Once the user has created a burn transaction on the source blockchain, he submits the event along with a proof to the \textsf{submit-event-proof} function. The proof can be either using the MMR approved by the federation or NIPoPoWs, depending on the deployment of the smart contract. Subsequently, he needs to wait for a contestation period of $k$ blocks. Any party who wishes to dispute the proof can do so within this period by calling the \textsf{submit-contesting-proof} function. After the period has passed, the user can finalize the event by calling the \textsf{finalize-event} method.
+
+\subsection{Decentralized Block Connection}
+We introduce a second way for proving and verifying the block connection which does not require a trusted federation. This is possible by making use of NIPoPoW infix proofs. The way the proving process works differs quite a bit from the federation, because in the federation there is a trusted source of truth -the federation majority- who vouch for the set of blocks on the source chain. Additionally, if a checkpoint is voted by the federation majority and a valid block connection proof based on that checkpoint is provided, it cannot be disputed.
+
+A NIPoPoW however can be disputed. For example, a malicious verifier could provide an infix proof for a block on a minority chain. In that case, a honest prover would have to step in and provide a proof for the best chain, which the verifier would then compare with the original proof to detect the fraud. For our decentralized approach, we assume a model where there exist two provers, one honest and one malicious similar to~\cite{nipopows,flyclient}. The block connection proof is a NIPoPoW infix proof for the block which includes the burn transaction.
```

Comment:
This is not understanable by someone not familiar with NIPoPoWs. Needs reworking to simplify and shorten. I don't think we need to talk about disputes at all here actually – this is not the topic of this paper.